rtengine/simpleprocess.cc

```diff
@@ -1361,58 +1367,47 @@ class ImageProcessor {
                 }
             }
 
-            delete tmplab;
-            tmplab = cropped;
-            cx = 0;
-            cy = 0;
-
+            tmplab.swap(cropped);
             params.crop.enabled = false;
         }
 
         assert(params.resize.enabled);
 
         // resize image
-        LabImage *resized = new LabImage(imw, imh);
-        ipf.Lanczos(tmplab, resized, tmpScale);
-        delete tmplab;
-        tmplab = resized;
+        {
+            std::unique_ptr<LabImage> resized(new LabImage(imw, imh));
+            ipf.Lanczos(tmplab.get(), resized.get(), scale_factor);
+            tmplab.swap(resized);
```

Comment:
@Floessie can you elaborate on why? http://www.cplusplus.com/reference/memory/unique_ptr/swap/